apollo-core/src/test/java/com/spotify/apollo/core/ServiceImplTest.java

```diff
@@ -446,6 +446,27 @@ public void testModuleClassesAreLifecycleManaged() throws Exception {
     assertThat(closed.get(), is(true));
   }
 
+  @Test
+  public void testChildModuleClassesAreLifecycleManaged() throws Exception {
+    AtomicBoolean created = new AtomicBoolean(false);
+    AtomicBoolean closed = new AtomicBoolean(false);
+    AtomicBoolean childCreated = new AtomicBoolean(false);
+    AtomicBoolean childClosed = new AtomicBoolean(false);
+    ModuleWithLifecycledKeys lifecycleModule =
+        new ModuleWithLifecycledKeys(created, closed, childCreated, childClosed);
+    Service service = ServiceImpl.builder("test")
+        .withModule(lifecycleModule)
+        .build();
+
+    try (Service.Instance instance = service.start()) {
+      instance.getSignaller().signalShutdown();
+      instance.waitForShutdown();
+    }
+
+    assertThat(childCreated.get(), is(true));
```

Comment:
nitty-pick: What do you think about repeating this at line 462? Something like this could be possible:

```
try (Service.Instance instance = service.start()) {
    assertThat(childCreated.get(), is(true));
    assertThat(childClosed.get(), is(false));
    instance.getSignaller().signalShutdown();  
    instance.waitForShutdown();    
}
```

From what I am looking in the code I think it should be fine as they are set in the moment you're creating the instance of the class. What do you think?